### ISSUE TITLE
off by 1 error line numbers and different error formatting under sbt on graalvm

### ISSUE THREAD

**Issue by aappddeevv** [lampepfl/dotty#7071]
Contrary to #340, I'm finding that source code reports for syntax errors are off by 1. This is on 0.17.

**Comment by smarter** [comment]
Can you provide an example ?

**Comment by aappddeevv** [comment]
I had tried reproducing with the g8 template but could not. What's funny is that the error messages look completely different between my project and the template eve though I'm running the same version of dotty (0.17-RC1) and dotty-sbt (0.3.3).

**Comment by aappddeevv** [comment]
Maybe its based on the features used in the file.

**Comment by aappddeevv** [comment]
Here's my error message:
```scala
0:1: Expected a toplevel definition
[error] error???
[error] ^
```
Here's the g8 template:
```scala
[info] Compiling 1 Scala source to /tmp/dotty-project-template/target/scala-0.17/classes ...
[error] -- [E080] Syntax Error: /tmp/dotty-project-template/src/main/scala/Main.scala:7:0 
[error] 7 |error?
[error]   |^^^^^
[error]   |Expected a toplevel definition
[error] one error found
```

**Comment by aappddeevv** [comment]
Oh, I think I found it. When running on graalvm, you get different error messages and the line numbers are off.

**Comment by smarter** [comment]
Does something similar happen when using Scala 2 instead of Dotty ?

**Comment by aappddeevv** [comment]
I don't have a scala2 proj handy to test that. I'll try to come across one.

**Comment by aappddeevv** [comment]
.@smarter believes that this is fixed by sbt/zinc#699.
